src/main/java/frc/robot/commands/drive/TurnDegrees.java

```diff
@@ -1,48 +1,56 @@
 package frc.robot.commands.drive;
 
-import static frc.robot.Constants.DriveConstants.TurnPID.*;
+import static frc.robot.Constants.DrivetrainConstants.TurnPID.*;
 
 import edu.wpi.first.math.controller.ProfiledPIDController;
 import edu.wpi.first.math.controller.SimpleMotorFeedforward;
 import edu.wpi.first.math.trajectory.TrapezoidProfile;
 import edu.wpi.first.wpilibj2.command.ProfiledPIDCommand;
-import frc.robot.subsystems.DriveSubsystem;
+import frc.robot.subsystems.DrivetrainSubsystem;
 import java.util.function.DoubleSupplier;
 
+/** A command that turns the robot the specified number of degrees. */
 public class TurnDegrees extends ProfiledPIDCommand {
-
+  // Feedforward - converts angular velocities to power outputs
   private static final SimpleMotorFeedforward m_feedforward =
       new SimpleMotorFeedforward(kS, kVDegreesPerSecond);
-  private DriveSubsystem m_drive;
 
-  /** Turns the specified number of degrees. */
-  public TurnDegrees(double goalAngle, DriveSubsystem drive) {
-    this(() -> goalAngle, drive);
+  private DrivetrainSubsystem m_drive;
+
+  /**
+   * Turns the specified number of degrees.
+   *
+   * @param degrees The number of degrees to turn
+   * @param drive The robot's drive subsystem
+   */
+  public TurnDegrees(double degrees, DrivetrainSubsystem drive) {
+    this(() -> degrees, drive);
   }
 
-  /** Turns the number of degrees supplied by goalAngle when this command is run. */
-  public TurnDegrees(DoubleSupplier goalAngle, DriveSubsystem drive) {
+  /**
+   * Turns the number of degrees supplied by angleSupplier when this command is run.
+   *
+   * @param angleSupplier A method to run in order to obtain the angle to turn to
+   * @param drive The robot's drive subsystem
+   */
+  public TurnDegrees(DoubleSupplier angleSupplier, DrivetrainSubsystem drive) {
     super(
         new ProfiledPIDController(
             kP,
             kI,
             kD,
             new TrapezoidProfile.Constraints(
                 kMaxVelocityDegreesPerSecond, kMaxAccelerationDegreesPerSecondSquared)),
-        drive::getHeading,
-        goalAngle,
+        drive::getAngle,
+        angleSupplier,
         (output, setpoint) ->
-            drive.arcadeDrive(0, output + m_feedforward.calculate(setpoint.velocity), false),
+            drive.arcadeDrive(0, -(output + m_feedforward.calculate(setpoint.velocity)), false),
```

Comment:
Do you know why this one is negative?

Reply:
The [`DifferentialDrive.arcadeDrive()` ](https://first.wpi.edu/wpilib/allwpilib/docs/release/java/edu/wpi/first/wpilibj/drive/DifferentialDrive.html#arcadeDrive(double,double)) method assumes that a clockwise rotation speed is positive, which is the opposite of what we use for getting the gyro angle. I think we got away with not negating it before because we inverted the wrong side of drive motors.